lib/contentful.js

```diff
@@ -75,20 +77,36 @@ export function createClient (params) {
     'X-Contentful-User-Agent': userAgentHeader
   }
 
-  const http = createHttpClient(axios, config)
+  function createContentfulApiForSpace (space, accessToken, environment) {
+    const httpClient = createHttpClient(axios, {
+      ...config,
+      headers: Object.assign({}, config.headers),
+      accessToken,
+      space
+    })
+    const getGlobalOptions = createGlobalOptions({
+      resolveLinks: config.resolveLinks,
+      environment,
+      removeUnresolved: config.removeUnresolved,
+      spaceBaseUrl: httpClient.defaults.baseURL,
+      environmentBaseUrl: `${httpClient.defaults.baseURL}environments/${environment}`
+    })
+    // Append environment to baseURL
+    httpClient.defaults.baseURL = getGlobalOptions().environmentBaseUrl
 
-  const getGlobalOptions = createGlobalOptions({
-    resolveLinks: config.resolveLinks,
-    environment: config.environment,
-    removeUnresolved: config.removeUnresolved,
-    spaceBaseUrl: http.defaults.baseURL,
-    environmentBaseUrl: `${http.defaults.baseURL}environments/${config.environment}`
-  })
-  // Append environment to baseURL
-  http.defaults.baseURL = getGlobalOptions().environmentBaseUrl
+    return createContentfulApi({
+      http: httpClient,
+      getGlobalOptions
+    })
+  }
+
+  const contentfulApiBySpaceId = Object.entries(config.additionalTokens).reduce((result, [spaceId, accessToken]) => {
```

Comment:
One improvement I can think of is comparing this access token to the access token provided for the main space and not calling the function if it is the case